src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.personrecord.api.controller
+
+import jakarta.validation.constraints.NotBlank
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.personrecord.api.controller.exceptions.PersonRecordNotFoundException
+import uk.gov.justice.digital.hmpps.personrecord.api.model.PersonIdentifierRecord
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+
+@RestController
+@RequestMapping("/api")
+@PreAuthorize("hasRole('ROLE_CORE_PERSON_RECORD_API__SEARCH__RO')")
+class SearchController(
+  private val personRepository: PersonRepository,
+) {
+
+  @GetMapping("/search/{personId}")
```

Comment:
Do I understand this correctly - we are providing an endpoint where you can pass any identifier and we will try to match it?
Surely the calling code knows what kind of identifier it is and could tell us? Something like this?
'/search/probation/{personId}' 
This seems like it will make (currently) 2 database calls which will never return anything.
Also, although it seems unlikely, if we end up with an identifier used by more than one source system in the future this will be a problem.

We are bringing complexity into our app by offering this endpoint - is it worth it?

Reply:
Discussed and updated 👍